server/pulp/plugins/util/publish_step.py

```diff
@@ -69,8 +74,9 @@ def __init__(self, step_type, status_conduit=None, non_halting_exceptions=None):
         self.last_report_time = 0
         self.last_reported_state = self.state
         self.timestamp = str(time.time())
-        self.non_halting_exceptions = non_halting_exceptions
+        self.non_halting_exceptions = non_halting_exceptions or []
```

Comment:
I think this would be better to be `self.non_halting_exceptions = non_halting_exceptions` and to change the signature and docstring to use `non_halting_exceptions=[]`.

It will have the same effect and will be more explicit in the docstring